contracts/examples/erc1155/src/lib.rs

```diff
@@ -0,0 +1,544 @@
+		}
+
+		// self.transfer_batch_event(&caller, &from, &to, ids, amounts);
+
+		Ok(())
+	}
+
+	#[endpoint(setApprovalForAll)]
+	fn set_approved_for_all(&self, operator: Address, approved: bool) {
+		let caller = self.get_caller();
+
+		self.set_is_approved(&operator, &caller, approved);
+
+		// self.approval_for_all_event(&caller, &operator, approved);
+	}
+
+	// returns assigned id
+	#[endpoint(createToken)]
+	fn create_token(&self, uri: &[u8], initial_supply: BigUint, is_fungible: bool) -> BigUint {
+		let big_uint_one = BigUint::from(1u32);
+
+		let creator = self.get_caller();
+		let type_id = self.get_last_valid_type_id() + big_uint_one.clone();
+
+		self.set_balance(&creator, &type_id, &initial_supply);
+		self.set_token_type_creator(&type_id, &creator);
+		self.set_is_fungible(&type_id, is_fungible);
+
+		if !is_fungible {
+			self.set_owner_for_range(&type_id, &big_uint_one, &initial_supply, &creator);
+			self.set_last_valid_token_id_for_type(&type_id, &initial_supply);
+		}
+
+		self.set_last_valid_type_id(&type_id);
+		self.set_token_type_uri(&type_id, uri);
+
+		// self.transfer_single_event(&caller, &from, &to, &id, &amount);
+
+		// uri event
+
+		type_id
+	}
+
+	#[endpoint]
+	fn mint(&self, type_id: BigUint, amount: BigUint) -> SCResult<()> {
+		let creator = self.get_token_type_creator(&type_id);
+
+		require!(
+			self.get_caller() == creator,
+			"Only the token creator may mint more tokens"
+		);
+
+		self.increase_balance(&creator, &type_id, &amount);
+
+		if self.get_is_fungible(&type_id) == false {
+			let last_valid_id = self.get_last_valid_token_id_for_type(&type_id);
+			let id_first = last_valid_id.clone() + BigUint::from(1u32);
+			let id_last = last_valid_id + amount;
+
+			self.set_owner_for_range(&type_id, &id_first, &id_last, &creator);
+
+			self.set_last_valid_token_id_for_type(&type_id, &id_last);
+		}
+
+		// self.transfer_single_event(&caller, &from, &to, &id, &amount);
+
+		Ok(())
+	}
+
+	#[endpoint]
+	fn burn(&self, type_id: BigUint, amount: BigUint) -> SCResult<()> {
+		require!(
+			self.get_is_fungible(&type_id) == true,
+			"Only fungible tokens can be burned"
+		);
+
+		let caller = self.get_caller();
+		let balance = self.balance_of(&caller, &type_id);
+
+		require!(balance >= amount, "Not enough tokens to burn");
+
+		self.decrease_balance(&caller, &type_id, &amount);
+
+		Ok(())
+	}
+
+	// views
+
+	#[view(supportsInterface)]
+	fn supports_interface(&self, interface_id: u32) -> bool {
+		interface_id == INTERFACE_SIGNATURE_ERC165 || interface_id == INTERFACE_SIGNATURE_ERC1155
+	}
+
+	#[view(balanceOf)]
+	fn balance_of(&self, owner: &Address, type_id: &BigUint) -> BigUint {
+		self.get_balance_mapper(&owner)
+			.get(&type_id)
+			.unwrap_or_else(|| BigUint::zero())
+	}
+
+	// returns balance for each (owner, id) pair
+	#[view(balanceOfBatch)]
+	fn balance_of_batch(&self, owners: &[Address], type_ids: &[BigUint]) -> SCResult<Vec<BigUint>> {
```

Comment:
Optional: can also provide them as argument pairs, i.e.
VarArgs<MultiArg2<Address, BigUint>>
might be easier to work with, but purely optional